CONTRIBUTING.md

```diff
@@ -1,35 +1,96 @@
 
-        (4) You are not expected to provide support for your contributions, except to the extent You desire to provide support. You may provide support for free, for a fee, or not at all. Unless required by applicable law or agreed to in writing, You provide your Contributions on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied, including, without limitation, any warranties or conditions of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A PARTICULAR PURPOSE.
+## How to Contribute
 
-then you just add a line saying
+Pull Requests (PRs) are the main and exclusive way to contribute to the project.
+In order for a PR to be accepted it needs to pass a list of requirements:
 
-    Signed-off-by: Random J Developer <random@developer.example.org>
+- If the PR is a bug fix, it has to include a new unit test that fails before the patch is merged.
+- If the PR is a new feature, it has to come with a suite of unit tests, that tests the new functionality.
+- In any case, all the PRs have to pass the personal evaluation of at least one of the [maintainers](MAINTAINERS).
 
-When committing using the command line you can sign off using the --signoff or -s flag. This adds a Signed-off-by line by the committer at the end of the commit log message.
 
-    git commit -s -m "Commit message"
+### Format of the commit message
+
+Every commit message should describe what was changed and, if applicable, the GitHub issue it relates to:
+
+```
+Skip argument validations for unknown capabilities. Fixes #623
+```
+
+The format can be described more formally as follows:
+
+```
+<what changed>. [Fixes #<issue-number>]
+```
+
+## Architecture
+
+The dashboard contains 2 parts:
+
+- Single page react application
+- Go web server to proxy and transform requests to bblfsd server
+
+For production usage we embed static javascript into go server using [go-bindata](https://github.com/jteeuwen/go-bindata).
+
+## Development
+
+The dashboard uses an intermediate API that connects to the bblfsh server and serves the dashboard front assets.
+
+Every time you change any source of the front assets, it is needed to regenerate the `server/asset/asset.go` containing the static files of the site.
+
+It can be done running
+```sh
+make assets
+```
+
+### Access the dashboard locally
+
+To run it locally you can run:
+```sh
+make serve
+```
+(Do it every time you modify something in the sources to re-generate the `server/asset/asset.go` file, the dashboard api, and to serve the updated dashboard itself)
+
+And access the dashboard through http://localhost:9999
+
+### Hot reloading mode
+
+You can also run frontend in hot reloading mode:
+
+```sh
+yarn start
+```
+
+Dashboard will be available on http://localhost:3000
+
+But it still requires go server on `9999` port to be available.
```

Comment:
To do so it can be run:

```shell
go install github.com/bblfsh/dashboard/server/...
bblfsh-dashboard -bblfsh-addr=0.0.0.0:9432
```

Reply:
better `go run server/cmd/bblfsh-dashboard/main.go`, not sure it worth to mention here though.

Reply:
I agree with the `go run` instead of the `go install + run it`. Way simpler.

About if it worth to mention it, I think it does. In this section, it is said how to run (for development purposes) the dashboard, and it is warned that the server is needed if it's proceded that way.
I think it worth to mention it to let the user launch the front and the back separately and explore its internals. Using `make serve` for this purpose is more confuse because it does things not really needed for this goal and way slower by the same reason.